apis/r/tests/testthat/helper-test-data.R

```diff
@@ -27,16 +27,22 @@ create_dense_matrix_with_int_dims <- function(nrows = 10, ncols = 5, seed = 1) {
   )
 }
 
-create_and_populate_soma_dataframe <- function(uri) {
-
-  arrow_schema <- arrow::schema(
+create_arrow_schema <- function() {
+  arrow::schema(
     arrow::field("foo", arrow::int32(), nullable = FALSE),
+    arrow::field("soma_joinid", arrow::int64(), nullable = FALSE),
     arrow::field("bar", arrow::float64(), nullable = FALSE),
     arrow::field("baz", arrow::large_utf8(), nullable = FALSE)
   )
+}
+
+create_and_populate_soma_dataframe <- function(uri) {
+
+  arrow_schema <- create_arrow_schema()
 
   tbl <- arrow::arrow_table(
     foo = 1L:10L,
+    soma_joinid = 1L:10L,
```

Comment:
kind of a nit but i like to make `soma_joinid` not startin at 0, just to demonstrate its flexibility as a sparse key, and to make sure we're not assuming anything about it being an array index per se anywhere inadvertently

Reply:
Good call. Changed in 70712ab0 .